src/nvim/message.c

```diff
@@ -137,6 +139,81 @@ int msg_attr(const char *s, const int attr) FUNC_ATTR_NONNULL_ARG(1)
   return msg_attr_keep((char_u *)s, attr, false);
 }
 
+int msg_echo_attr(const char *s, const int attr) FUNC_ATTR_NONNULL_ARG(1)
```

Comment:
Do not put FUNC_ATTR on the same line as a function name, this variant both visually hides attributes and produces unneccessary line change if something needs to be added (e.g. new argument or another attribute).

Reply:
BTW, is this a function intended to be used in `ex_echo`? It is not good to have two implementations doing essentially the same task; also using function there makes testing a lot easier and allows CI to catch errors before test suite is good enough.

Reply:
I did not intend to use it in `ex_echo`, but I guess I could modify `ex_echo` to use it (if it is the purpose of the question)